jenkins/pipelines/cd/tidb-operator.groovy

```diff
@@ -0,0 +1,409 @@
+  tolerations:
+  - effect: NoSchedule
+    key: tidb-operator
+    operator: Exists
+'''
+final dockerSyncYaml = '''
+apiVersion: v1
+kind: Pod
+metadata:
+  name: regctl
+spec:
+  containers:
+  - name: regctl
+    image: hub.pingcap.net/jenkins/regctl
+    args: ["sleep", "infinity"]
+  tolerations:
+  - effect: NoSchedule
+    key: tidb-operator
+    operator: Exists
+'''
+final uploaderYaml = '''
+apiVersion: v1
+kind: Pod
+metadata:
+  name: uploader
+spec:
+  containers:
+  - name: uploader
+    image: hub.pingcap.net/jenkins/uploader
+    args: ["sleep", "infinity"]
+  - name: helm
+    image: hub.pingcap.net/jenkins/helm:2.14.1
+    args: ["sleep", "infinity"]
+  tolerations:
+  - effect: NoSchedule
+    key: tidb-operator
+    operator: Exists
+'''
+
+
+pipeline {
+    agent none
+    parameters {
+        string(name: 'GitRef', defaultValue: 'master', description: 'branch or commit hash')
+        string(name: 'ReleaseTag', defaultValue: 'test', description: 'empty means the same with GitRef')
+    }
+    stages {
+        stage("PARAMS") {
+            steps {
+                script {
+                    ReleaseTag = params.ReleaseTag
+                    if (!ReleaseTag) {
+                        ReleaseTag = params.GitRef
+                    }
+                    println("ReleaseTag: $ReleaseTag")
+                }
+            }
+        }
+        stage("BUILD") {
+            stages {
+                stage("Bin") {
+                    agent {
+                        kubernetes {
+                            yaml goBuildYaml
+                            defaultContainer 'builder'
+                            cloud "kubernetes-ng"
+                            namespace "jenkins-tidb-operator"
+                        }
+                    }
+                    stages {
+                        stage("checkout") {
+                            steps {
+                                script {
+                                    sh '''curl -O http://fileserver.pingcap.net/download/tmp/tidb-operator.tar.gz
+                                          tar -xzf tidb-operator.tar.gz
+                                          rm tidb-operator.tar.gz'''
+                                    def scmVars = checkout changelog: false, poll: false, scm: [
+                                            $class           : 'GitSCM',
+                                            branches         : [[name: "${params.GitRef}"]],
+                                            userRemoteConfigs: [[
+                                                                        refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pull/*',
+                                                                        url    : 'https://github.com/pingcap/tidb-operator.git',
+                                                                ]]
+                                    ]
+                                    sh '''git clean -f .'''
```

Comment:
why need this step, checkout step has a option to clean git dir

Reply:
go mod download may change go.sum, causing version tag contains `-dirty`